File: tests/npy.rs
use automesh::Npy;

const NELX: usize = 3;
const NELY: usize = 5;
const NELZ: usize = 4;

const GOLD: [[[u8; NELX]; NELY]; NELZ] = [
    [[1, 1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1]],
    [[1, 1, 1], [1, 0, 0], [1, 1, 0], [1, 0, 0], [1, 0, 0]],
    [[1, 1, 1], [1, 0, 0], [1, 1, 0], [1, 0, 0], [1, 0, 0]],
    [[1, 1, 1], [1, 0, 0], [1, 1, 0], [1, 0, 0], [1, 0, 0]],
];

#[test]
fn read_npy() {
    let npy = Npy::new("tests/npy/f.npy");
    GOLD.iter()
        .zip(npy.get_data().outer_iter())
        .for_each(|(gold_i, spn_i)| {
            gold_i
                .iter()
                .zip(spn_i.outer_iter())
                .for_each(|(gold_ij, spn_ij)| {
                    gold_ij
                        .iter()
                        .zip(spn_ij.iter())
                        .for_each(|(gold_ijk, spn_ijk)| assert_eq!(gold_ijk, spn_ijk))
                })
        })
}

File: tests/npy.py
import numpy as np
from automesh import Npy

gold = np.array([
    [[1, 1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1], [1, 1, 1]],
    [[1, 1, 1], [1, 0, 0], [1, 1, 0], [1, 0, 0], [1, 0, 0]],
    [[1, 1, 1], [1, 0, 0], [1, 1, 0], [1, 0, 0], [1, 0, 0]],
    [[1, 1, 1], [1, 0, 0], [1, 1, 0], [1, 0, 0], [1, 0, 0]],
], dtype='uint8')


def test_new():
    npy = Npy('tests/npy/f.npy')
    assert (npy.get_data() == gold).all()


def test_temporary():
    npy = Npy('tests/npy/f.npy')
    exo = npy.exodus()
    print(exo.element_connectivity)
    print("Temporary test.")
